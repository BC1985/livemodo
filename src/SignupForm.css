form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

#signup-form input,
textarea {
  background-color: #cccc;
  padding: 5px;
  display: inline-block;
  margin-left: 100px;
}

button {
  align-self: center;
  margin: 20px;
}

.text-input {
  display: inline-block;
}

#sign-form label {
  margin-right: 100px;
}
