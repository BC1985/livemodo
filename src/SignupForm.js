import React from "react";

function SignupForm() {
  const spanStyle = {
    color: "gray",
    fontStyle: "italic",
    fontSize: "12px"
  };
  return (
    <>
      <div className="call-to-action">
        <h3>Join the Livemodo community!</h3>
      </div>
      <div className="signup-form">
        <form id="signup">
          <div className="form-text-input">
            <label>*First Name</label>
            <input type="text" placeholder="e.g Tommy" required />
          </div>
          <div className="form-text-input">
            <label>*Last Name</label>
            <input type="text" placeholder="e.g Wisseau" required />
          </div>
          <div className="form-text-input">
            <label>*Username</label>
            <input type="text" required />
          </div>
          <div className="form-text-input">
            <label>*Email</label>
            <input type="text" placeholder="e.g tommy@theroom.com" required />
          </div>
          <div className="form-text-input">
            <label>*Password</label>
            <input type="text" required />
          </div>
          <p style={spanStyle}>(* indicates required field)</p>
          <p>
            Pick the option that best describes your concergoing habits{" "}
            <span style={spanStyle}>(optional)</span>
          </p>
          <div className="form-buttons">
            <div>
              <input type="radio" value="homebody" />
              Homebody
            </div>
            <div>
              <input type="radio" />
              Occasional concertgoer
            </div>
            <div>
              <input type="radio" />
              Concert Connuiseur
            </div>
            <button type="submit">Sign me up!</button>
          </div>
        </form>
      </div>
    </>
  );
}

export default SignupForm;
