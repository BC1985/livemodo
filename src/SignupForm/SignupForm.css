.signup-container {
  color: whitesmoke;
  padding-bottom: 20%;
}

.call-to-action > h2 {
  padding: 10% 0 5% 0;
  margin: 0;
}

.error {
  border: 2px solid red;
}
#register-button:enabled {
  display: flex;
  display: inline-block;
  margin: 5% auto 0;
  height: 25px;
  width: 100%;
  border: 1px solid #6e7175;
  text-shadow: -1px -1px 0 rgba(0, 0, 0, 0.3);

  border-radius: 5px;
  color: #fff;
  background-image: -webkit-gradient(
    linear,
    left top,
    left bottom,
    from(#646d7e),
    to(#22242e)
  );
  cursor: pointer;
}
#register-button:enabled:hover {
  color: #fff;
  background-image: -webkit-gradient(
    linear,
    left top,
    left bottom,
    from(#373f51),
    to(#060c0f)
  );
  cursor: pointer;
}
#register-button:disabled {
  display: flex;
  display: inline-block;
  margin: 5% auto 0;
  height: 25px;
  width: 100%;
  border: none;
  border-radius: 5px;
  background-color: gray;
  color: rgb(197, 195, 195);
}
/* ----------- iPhone 5, 5S, 5C and 5SE ----------- */
/* Portrait */
@media only screen and (min-device-width: 320px) and (max-device-width: 568px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: portrait) {
  .signup-container {
    background: url(../images/Resized/stage-small-B&W.png);
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
  }
  .signup-form {
    padding: 0 10%;
  }
  #signup {
    background-color: rgba(39, 37, 37, 0.6);
    border-radius: 10px;
    padding: 15px;
  }

  #signup input[type="text"] {
    border-radius: 3px;

    background-color: #cccc;
    padding: 5px;
    width: 100%;
    display: inline-block;
    color: black;
  }

  #button {
    align-self: center;
  }

  #signup label {
    display: inline-block;
    padding: 5px 0 5px 0;
  }
}

/* Landscape */
@media only screen and (min-device-width: 320px) and (max-device-width: 568px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: landscape) {
  .signup-container {
    background: url(../images/Resized/stage-small-B&W.png);
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
  }
  .signup-form {
    padding: 0 10%;
  }
  #signup {
    background-color: rgba(39, 37, 37, 0.6);
    border-radius: 10px;
    padding: 15px;
  }

  #signup input[type="text"] {
    border-radius: 3px;

    background-color: #cccc;
    padding: 5px;
    width: 100%;
    display: inline-block;
    color: black;
  }

  #button {
    align-self: center;
  }

  #signup label {
    display: inline-block;
    padding: 5px 0 5px 0;
  }
}

/* ----------- iPhone 6, 6S, 7 and 8 ----------- */

/* Portrait */
@media only screen and (min-device-width: 375px) and (max-device-width: 667px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: portrait) {
  .signup-container {
    background: url(../images/Resized/stage-small-B&W.png);
    background-size: cover;
    background-position: center;
    padding-bottom: 20%;
  }
  .signup-form {
    padding: 0 20%;
  }
  #signup {
    background-color: rgba(39, 37, 37, 0.6);
    border-radius: 10px;
    padding: 15px;
  }

  #signup input[type="text"] {
    border-radius: 3px;

    background-color: #cccc;
    padding: 5px;
    width: 100%;
    display: inline-block;
    color: black;
  }

  #button {
    align-self: center;
  }

  #signup label {
    display: inline-block;
    padding: 5px 0 5px 0;
  }
}

/* Landscape */
@media only screen and (min-device-width: 375px) and (max-device-width: 667px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: landscape) {
  .signup-container {
    background: url(../images/Resized/stage-small-B&W.png);
    background-size: cover;
    background-position: center;
  }
  .signup-form {
    padding: 0 20%;
  }
  #signup {
    background-color: rgba(39, 37, 37, 0.6);
    border-radius: 10px;
    padding: 15px;
  }

  #signup input[type="text"] {
    border-radius: 3px;

    background-color: #cccc;
    padding: 5px;
    width: 100%;
    display: inline-block;
    color: black;
  }

  #button {
    align-self: center;
  }

  #signup label {
    display: inline-block;
    padding: 5px 0 5px 0;
  }
}

/* ----------- iPhone 6+, 7+ and 8+ ----------- */

/* Portrait */
@media only screen and (min-device-width: 414px) and (max-device-width: 736px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: portrait) {
  .signup-container {
    padding-bottom: 20%;
  }
}

/* Landscape */
@media only screen and (min-device-width: 414px) and (max-device-width: 736px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: landscape) {
  .signup-container {
    background: url(../images/Resized/stage-small-B&W.png);
    background-size: cover;
    background-position: center;
  }
  .signup-form {
    padding: 0 20%;
  }
  #signup {
    background-color: rgba(39, 37, 37, 0.6);
    border-radius: 10px;
    padding: 15px;
  }

  #signup input[type="text"] {
    border-radius: 3px;

    background-color: #cccc;
    padding: 5px;
    width: 100%;
    display: inline-block;
    color: black;
  }

  #button {
    align-self: center;
  }

  #signup label {
    display: inline-block;
    padding: 5px 0 5px 0;
  }
}

/* ----------- iPhone X ----------- */

/* Portrait */
@media only screen and (min-device-width: 375px) and (max-device-width: 812px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: portrait) {
  .signup-container {
    background: url(../images/Resized/stage-small-B&W.png);
    background-size: cover;
    background-position: center;
    padding: 10% 10% 30%;
  }
  .signup-form {
    padding: 0 10%;
  }
  #signup {
    background-color: rgba(39, 37, 37, 0.6);
    border-radius: 10px;
  }
}

/* Landscape */
@media only screen and (min-device-width: 375px) and (max-device-width: 812px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: landscape) {
  .signup-container {
    background: url(../images/Resized/stage-small-B&W.png);
    background-size: cover;
    background-position: center;
  }
  .signup-form {
    padding: 0 15%;
  }
  #signup {
    background-color: rgba(39, 37, 37, 0.6);
    border-radius: 10px;
    padding: 15px;
  }
  #signup input[type="text"] {
    border-radius: 3px;

    background-color: #cccc;
    padding: 5px;
    width: 100%;
    display: inline-block;
    color: black;
  }
}
/* ----------- iPad 3, 4 and Pro 9.7" ----------- */

/* Portrait */
@media only screen and (min-device-width: 768px) and (max-device-width: 1024px) and (orientation: portrait) and (-webkit-min-device-pixel-ratio: 2) {
  .signup-container {
    background: url(../images/Resized/stage-small-B&W.png);
    background-size: cover;
    background-position: center;
    padding: 0 20% 30%;
  }
  #signup {
    background-color: rgba(39, 37, 37, 0.6);
    border-radius: 10px;
    padding: 15px;
  }
  #signup input[type="text"] {
    border-radius: 3px;
    background-color: #cccc;
    padding: 10px;
    font-size: 1em;
    margin: 10px 0;
    width: 100%;
    display: inline-block;
    color: black;
  }
}

/* Landscape */
@media only screen and (min-device-width: 768px) and (max-device-width: 1024px) and (orientation: landscape) and (-webkit-min-device-pixel-ratio: 2) {
  .signup-container {
    background: url(../images/Resized/stage-small-B&W.png);
    background-size: cover;
    background-position: center;
  }
  .signup-form {
    padding: 0 15%;
  }
  #signup {
    background-color: rgba(39, 37, 37, 0.6);
    border-radius: 10px;
    padding: 15px;
  }
  #signup input[type="text"] {
    border-radius: 3px;
    font-size: 1em;

    background-color: #cccc;
    padding: 10px;
    margin: 10px 0;
    width: 100%;
    display: inline-block;
    color: black;
  }
}

/* ----------- Laptops and desktops ------------ */

/* ----- Non-Retina Screens ---- */
@media screen and (min-device-width: 1200px) and (max-device-width: 1600px) and (-webkit-min-device-pixel-ratio: 1) {
  .signup-container {
    background: url(../images/Resized/stage-small-B&W.png);
    background-size: cover;
    background-position: center;
  }
  .call-to-action > h2 {
    padding-top: 10%;
  }
  .signup-form {
    padding: 0 10%;
  }
  #signup {
    background-color: rgba(39, 37, 37, 0.6);
    border-radius: 10px;
    /* padding: 15px; */
  }
  #signup input[type="text"] {
    border-radius: 3px;

    background-color: #cccc;
    padding: 10px;
    margin: 10px 0;
    width: 100%;
    display: inline-block;
    color: black;
  }
}

/* ----- Retina Screens ------ */
@media screen and (min-device-width: 1200px) and (max-device-width: 1600px) and (-webkit-min-device-pixel-ratio: 2) and (min-resolution: 192dpi) {
  .signup-container {
    background: url(../images/Resized/stage-small-B&W.png);
    background-size: cover;
    background-position: center;
  }
  .signup-form {
    padding: 0 15%;
  }
  #signup {
    background-color: rgba(39, 37, 37, 0.6);
    border-radius: 10px;
    padding: 15px;
  }
  #signup input[type="text"] {
    border-radius: 3px;

    background-color: #cccc;
    padding: 10px;
    margin: 10px 0;
    width: 100%;
    display: inline-block;
    color: black;
  }
}

@media screen and (min-device-width: 1600px) and (-webkit-min-device-pixel-ratio: 1) {
  .signup-container {
    background: url(../images/Resized/stage-small-B&W.png);
    background-size: cover;
    background-position: center;
  }
  .signup-form {
    padding: 0 25%;
  }

  #signup {
    font-size: 0.7em;
    background-color: rgba(39, 37, 37, 0.6);
    border-radius: 10px;
    padding: 15px;
    width: 50%;
    margin: 0 auto;
  }
  label {
    display: block;
  }
  #signup input[type="text"] {
    border-radius: 3px;
    font-size: 0.8em;
    justify-content: center;
    background-color: #cccc;
    margin: 5px 0;
    width: 100%;
    display: inline-block;
    color: black;
  }
  #required-fields {
    font-size: 0.8em;
  }
  #signup > button {
    font-size: 0.8em;
    display: inline-block;
  }
}
