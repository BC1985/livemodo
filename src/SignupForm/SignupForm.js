import React, { Component } from "react";
import "./SignupForm.css";
import AuthApiService from "../services/auth-api-service";
import { withRouter } from "react-router-dom";
import { validateRegisration, shouldBeError } from "../Validation/validation";


import { useState } from "react";

function SignupForm(props) {
  const [input, setInput] = useState({
    first_name: "",
    last_name: "",
    username: "",
    email: "",
    password: "",
  });
  const { email, password, first_name, last_name, username } = input;
  const [touched, setTouched] = useState({
    first_name: false,
    last_name: false,
    username: false,
    email: false,
    password: false,
  });
  const [errors, setErrors] = useState({
    first_name: false,
    last_name: false,
    username: false,
    email: false,
    password: false,
  });
  const [passwordError, setPasswordError] = useState(false)
  const [hasError, setHasError] = useState(false)
  const [error, setError] = useState("")
  // state = {
  //   error: null,


  //   errorMessage: false,
  //   passwordError: false
  // };

  const onChange = e => {
    const { name, value } = e.target;
    setInput(input => ({ ...input, [name]: value }));
  };

  const validatePassword = () => {
    
    const REGEX_UPPER_LOWER_NUMBER_SPECIAL = /(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])[\S]+/;
    if (password.length < 8) {
      return "Password must be at least 8 characters";
    }
    if (password.length > 72) {
      return "Password must be less than 72 characters";
    }
    if (password.startsWith(" ") || password.endsWith(" ")) {
      return "Password must not start or end with empty spaces";
    }
    if (!REGEX_UPPER_LOWER_NUMBER_SPECIAL.test(password)) {
      return "Password must contain at least one upper case, lower case, number and special character";
    }
    return null;
  };

  const renderError = e => {
    setPasswordError(true)
  };

  const handleSubmit = e => {
    e.preventDefault();
    const passwordNotValid = validatePassword();

    if (passwordNotValid) {
      // this.setState({ error: true }, renderError());
      setHasError(true)
      renderError()
    } else {
      // const { first_name, last_name, email, username, password } = e.target;
      AuthApiService.postUser({
        first_name,
        last_name,
        username,
        password,
        email
      })
        .then(user => {
          first_name.value = "";
          last_name.value = "";
          username.value = "";
          password.value = "";
          email.value = "";
        })
        .catch(console.log(res));

      props.history.push("thank-you");
    }
  };
  const handleBlur = field => e => {
    setTouched({ ...touched, [field]: true });
  };

  const spanStyle = {
    color: "whiteSmoke",
    fontStyle: "italic",
    fontSize: "0.8em",
  };
  const ifPasswordError = {
    border: "2px solid red",
  };

  const validationErrors = validateRegisration(
    first_name,
    last_name,
    username,
    password,
    email
  );
  const isEnabled = !Object.keys(validationErrors).some(x => errors[x]);
  
  const errorMessage = validatePassword();
  return (
    <div className="signup-container">
      <div className="call-to-action">
        <h2>Join the Livemodo community!</h2>
      </div>
      <div className="signup-form">
        <form id="signup" onSubmit={handleSubmit}>
          <div className="text-input">
            <label>*First Name </label>
            <input
              className={
                shouldBeError("first_name", validationErrors, touched) ? "error" : null
              }
              type="text"
              name="first_name"
              placeholder="e.g Tommy"
              value={first_name}
              onBlur={handleBlur("first_name", validationErrors, touched)}
              onChange={onChange}
              required
            />
          </div>
          <div className="text-input">
            <label>*Last Name </label>
            <input
              className={
                shouldBeError("last_name", validationErrors, touched) ? "error" : null
              }
              type="text"
              name="last_name"
              placeholder="e.g Wisseau"
              value={last_name}
              onBlur={handleBlur("last_name", validationErrors, touched)}
              onChange={onChange}
              required
            />
          </div>
          <div className="text-input">
            <label>*Username </label>
            <input
              className={
                shouldBeError("username", validationErrors, touched) ? "error" : null
              }
              type="text"
              name="username"
              value={username}
              onBlur={handleBlur("username", validationErrors, touched)}
              onChange={onChange}
              required
            />
          </div>
          <div className="text-input">
            <label>*Email </label>
            <input
              className={
                shouldBeError("email", validationErrors, touched) ? "error" : null
              }
              type="text"
              name="email"
              placeholder="e.g tommy@theroom.com"
              value={email}
              onBlur={handleBlur("email", validationErrors, touched)}
              onChange={onChange}
              required
            />
          </div>
          <div className="text-input">
            <label>*Password </label>
            <input
              className={
                shouldBeError("password", validationErrors, touched) ? "error" : null
              }
              type="text"
              name="password"
              value={password}
              onBlur={handleBlur("password", validationErrors, touched)}
              onChange={onChange}
              required
              style={passwordError ? ifPasswordError : null}
            />
            <span className="errorMessage">
              {error ? errorMessage : ""}
            </span>
          </div>

          <p style={spanStyle}>* indicates required field</p>
          <button
            id="register-button"
            type={!isEnabled ? "disabled" : "submit"}
            disabled={!isEnabled}
          >
            Submit
          </button>
        </form>
        <div className="register-push" />
      </div>
    </div>
  );
}

export default withRouter(SignupForm);
