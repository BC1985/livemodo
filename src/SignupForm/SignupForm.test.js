import React from "react";
import ReactDOM from "react-dom";
import SignupForm from "../SignupForm/SignupForm";
import { MemoryRouter } from "react-router-dom";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <SignupForm />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});
