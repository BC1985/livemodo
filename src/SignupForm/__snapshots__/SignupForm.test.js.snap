// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as expected 1`] = `
<div
  className="signup-container"
>
  <div
    className="call-to-action"
  >
    <h2>
      Join the Livemodo community!
    </h2>
  </div>
  <div
    className="signup-form"
  >
    <form
      id="signup"
      onSubmit={[Function]}
    >
      <div
        className="text-input"
      >
        <label>
          *First Name 
        </label>
        <input
          className={null}
          name="first_name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="e.g Tommy"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="text-input"
      >
        <label>
          *Last Name 
        </label>
        <input
          className={null}
          name="last_name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="e.g Wisseau"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="text-input"
      >
        <label>
          *Username 
        </label>
        <input
          className={null}
          name="username"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="text-input"
      >
        <label>
          *Email 
        </label>
        <input
          className={null}
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="e.g tommy@theroom.com"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="text-input"
      >
        <label>
          *Password 
        </label>
        <input
          className={null}
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          style={null}
          type="text"
          value=""
        />
        <span
          className="errorMessage"
        >
          
        </span>
      </div>
      <p
        style={
          Object {
            "color": "whiteSmoke",
            "fontSize": "0.8em",
            "fontStyle": "italic",
          }
        }
      >
        * indicates required field
      </p>
      <button
        disabled={true}
        id="register-button"
        type="disabled"
      >
        Submit
      </button>
    </form>
    <div
      className="register-push"
    />
  </div>
</div>
`;
